src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
+            var compilation = (CSharpCompilation)_semanticModel.Compilation;
+            IOperation conditionOp;
+            if (parameter.Type.IsNullableType())
+            {
+                conditionOp = new UnaryOperation(
+                    UnaryOperatorKind.Not,
+                    new InvocationOperation(
+                        targetMethod: ((IMethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue)).Construct(parameter.Type),
+                        instance: new ParameterReferenceOperation(parameter, _semanticModel, syntax, parameter.Type, constantValue, isImplicit: true),
+                        isVirtual: false,
+                        arguments: ImmutableArray<IArgumentOperation>.Empty,
+                        _semanticModel,
+                        syntax,
+                        compilation.GetSpecialType(SpecialType.System_Boolean),
+                        constantValue,
+                        isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    operatorMethod: null,
+                    _semanticModel,
+                    syntax,
+                    compilation.GetSpecialType(SpecialType.System_Boolean),
+                    constantValue,
+                    isImplicit: true);
+            }
+            else
+            {
+                conditionOp = new BinaryOperation(BinaryOperatorKind.Equals,
+                    new ParameterReferenceOperation(parameter, _semanticModel, syntax, parameter.Type, constantValue, isImplicit: true),
+                    new LiteralOperation(_semanticModel, syntax, parameter.Type, ConstantValue.Null, isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    isCompareText: false,
+                    operatorMethod: null,
+                    unaryOperatorMethod: null,
+                    _semanticModel,
+                    syntax,
+                    compilation.GetSpecialType(SpecialType.System_Boolean),
+                    constantValue,
+                    isImplicit: true);
+            }
+
+            var argumentNullExceptionMethodSymbol = (IMethodSymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_ArgumentNullException__ctorString);
+            var argumentNullExceptionType = compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException);
+            var argumentNullExceptionObject = new ObjectCreationOperation(
+                        constructor: argumentNullExceptionMethodSymbol,
+                        initializer: new ObjectOrCollectionInitializerOperation(
+                                initializers: ImmutableArray<IOperation>.Empty,
+                                _semanticModel,
+                                syntax,
+                                compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException),
```

Comment:
>compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException) [](start = 32, length = 72)

`argumentNullExceptionType`

Reply:
The entire initializer should be null, but as an FYI initializers themselves do not have a type, so this would properly be `null` if it was sticking around.

---
In reply to: [305537111](https://github.com/dotnet/roslyn/pull/37381#discussion_r305537111) [](ancestors = 305537111)